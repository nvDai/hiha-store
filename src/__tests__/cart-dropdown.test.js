import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import CartDropdown from '../components/cart-dropdown/cart-dropdown.component'

describe('CartDropdown', () => {
	let wrapper, mockProps

	beforeEach(() => {
		mockProps = {
			cartItems: [],
			hideCart: jest.fn(),
		}
		wrapper = shallow(<CartDropdown {...mockProps} />)
	})

	it('should render without crashing', () => {
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('should render CartItem list', () => {
		mockProps = {
			...mockProps,
			cartItems: [
				{
					id: 1,
					imageUrl:
						'https://ik.imagekit.io/fireman35/images/ZYW3VTp/brown-brim.png',
					name: 'Brown Brim',
					price: 25,
					quantity: 2,
				},
				{
					id: 2,
					imageUrl:
						'https://ik.imagekit.io/fireman35/images/ypkgK0X/blue-beanie.png',
					name: 'Blue Beanie',
					price: 18,
					quantity: 2,
				},
			],
		}
		wrapper = shallow(<CartDropdown {...mockProps} />)
		expect(wrapper.find('.cart-items').props().children.length).toEqual(2)
	})

	it('should render empty message', () => {
		expect(wrapper.find('.empty-message').length).toEqual(1)
		// expect(wrapper1.find('CustomButton').length).toEqual(1)
	})

	describe('CustomButton', () => {
		it('should hide cart when click', () => {
			wrapper.find('CustomButton').simulate('click')
			expect(mockProps.hideCart).toHaveBeenCalledTimes(1)
		})
	})
})
