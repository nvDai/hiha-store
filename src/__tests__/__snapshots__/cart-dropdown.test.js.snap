// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown should render without crashing 1`] = `
<div
  className="cart-dropdown"
>
  <div
    className="cart-items"
  >
    <span
      className="empty-message"
    >
      Your cart is empty
    </span>
  </div>
  <CustomButton
    onClick={[MockFunction]}
  >
    GO TO CHECKOUT
  </CustomButton>
</div>
`;
